File: a010.cpp
#include<iostream>
#include<cstdio>
using namespace std;
int main(){
	int input;
	cin>>input;
	int idx=2,m;
	bool fir=1;
	while(input>1){
		m=0;
		while(input%idx==0){
			input/=idx;
			m++;
		}
		if(m==0){
			idx++;
			continue;
		}
		else{
			if(fir)fir=0;
			else cout<<" * ";
			if(m==1)cout<<idx;
			else printf("%d^%d",idx,m);
		}
		idx++;
	}
	cout<<endl;
}

File: a010.py
n = int(input())
for i in range(2,100000000):
	cnt=0
	while n%i==0:
		n/=i
		cnt+=1
	if cnt==1:
		print(str(i),end='')
	elif cnt>1:
		print(str(i)+"^"+str(cnt),end='')
	if n!=1 and cnt!=0 :
		print(" * ",end='')
	if n==1 :
		print()
		break